### DISCUSSION TITLE
Filtering on relational field?

### DISCUSSION THREAD

**Discussion by AaronBeaudoin** [directus/directus#11905] [Archive]
Is it possible to filter on a relational field?

Suppose I have a data model with **products** and **product variants**, a pretty standard requirement. Each product has a relational field to a **product option**, which itself has a relational field to a **product option value**. So for example, a product option might be "Color" and a value might be "Black".

The product variant by definition relates to a specific combination of said option values, so I create a M2M field on the variant which should relate to items in the product option value collection. However, it would be incorrect data if the M2M field contained option values from an option on a **different product**. Therefore I want to give the M2M field a filter that says the product of the variant (specified via a M2O field "Product" on the **variant**) should match the product of the option value (specified via a M2O field "Product" on the **option value's parent option**).

So in the filter for the M2M field "Option Values" on the variant, I'm looking at something like this:

<img width="365" alt="Screen Shot 2022-03-01 at 10 37 26 AM" src="https://user-images.githubusercontent.com/26592486/156210331-3196060f-497e-4c04-9a5d-125dc995ea9a.png">
 
So here I would like to have the value be something like `$THIS.product.id` or `$SELF.product.id`, where `$THIS` or `$SELF` is the current product variant item. How would I accomplish this sort of constraint/filtering in Directus?

**Comment by rijkvanzanten** [comment]
There's currently no native way in the filter structure to use a value of the current parent item in the filter rules. There's a discussion here regarding subqueries in the filter setup: https://github.com/directus/directus/discussions/10733#discussioncomment-2133055, and another one here around supporting a new $VALUE dynamic variable that would allow for this: https://github.com/directus/directus/discussions/2993

**Comment by snimavat** [comment]
@AaronBeaudoin how did you endup overcoming this limitation?

I m Trying to do some thing similar, and for the exact use case of product variants.
My requirement is "What variants can be selected for a product depends on which category the product belongs to"
For example, color for clothes and dimension for furniture category, etc. Available variants are specified at category level
and they are selected when creating product variant

**Reply by marcozac** [reply]
It seems you can do that using double brackets with the name of the field

<img width="854" alt="Screenshot 2024-09-01 at 09 27 06" src="https://github.com/user-attachments/assets/a3e702dd-c9a3-495c-82bb-ca2ed289b639">
